tealer/__main__.py

```diff
@@ -47,40 +182,98 @@ class ListDetectors(argparse.Action):  # pylint: disable=too-few-public-methods
     def __call__(
         self, parser: argparse.ArgumentParser, *args: Any, **kwargs: Any
     ) -> None:  # pylint: disable=signature-differs
-        detectors = get_detectors()
+        detectors, _ = get_detectors_and_printers()
         output_detectors(detectors)
         parser.exit()
 
 
-def get_detectors() -> List[Type[AbstractDetector]]:
-    detectors = [getattr(all_detectors, name) for name in dir(all_detectors)]
-    detectors = [d for d in detectors if inspect.isclass(d) and issubclass(d, AbstractDetector)]
-    return detectors
+class ListPrinters(argparse.Action):  # pylint: disable=too-few-public-methods
+    def __call__(
+        self, parser: argparse.ArgumentParser, *args: Any, **kwargs: Any
+    ) -> None:  # pylint: disable=signature-differs
+        _, printers = get_detectors_and_printers()
+        output_printers(printers)
+        parser.exit()
+
+
+def get_detectors_and_printers() -> Tuple[
+    List[Type[AbstractDetector]], List[Type[AbstractPrinter]]
+]:
+    detector_classes = [getattr(all_detectors, name) for name in dir(all_detectors)]
+    detector_classes = [
+        d for d in detector_classes if inspect.isclass(d) and issubclass(d, AbstractDetector)
+    ]
 
+    printer_classes = [getattr(all_printers, name) for name in dir(all_printers)]
+    printer_classes = [
+        d for d in printer_classes if inspect.isclass(d) and issubclass(d, AbstractPrinter)
+    ]
 
-def main() -> None:
+    return detector_classes, printer_classes
 
-    args = parse_args()
 
-    with open(args.program, encoding="utf-8") as f:
-        print(f"Analyze {args.program}")
-        teal = parse_teal(f.read())
+def main() -> None:  # pylint: disable=too-many-branches
```

Comment:
Maybe we can split main into sub functions to avoid the too many branches issue?

like
- `handle_print_cfg`
- `handle_detector_and_printers` (register & run)
- `handle_output`
- ...

Reply:
yes, this will also make reading code easy